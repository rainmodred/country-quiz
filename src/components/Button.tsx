/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { colors } from '../styles/colors';

interface ButtonProps {
  children: React.ReactNode;
  primary?: boolean;
  onClick?: () => void;
}

export default function Button({ children, primary, onClick }: ButtonProps) {
  return (
    <button
      onClick={onClick}
      type="button"
      css={css`
        border-radius: 12px;
        font-size: 18px;
        line-height: 27px;
        font-weight: 700;
        border: none;
        cursor: pointer;
        transition: all 0.2s ease-in;
        padding: ${primary ? '15px 36px 14px 37px' : ' 18px 61px 17px 61px'};
        border: ${primary ? 'none' : `2px solid ${colors.colorResult}`};
        color: ${primary ? colors.colorWhite : colors.colorResult};
        background: ${primary ? colors.colorHover : 'none'};
        box-shadow: ${primary ? `0px 2px 4px ${colors.colorShadow}` : 'none'};
        &:hover {
          opacity: 0.7;
        }
      `}
    >
      {children}
    </button>
  );
}
