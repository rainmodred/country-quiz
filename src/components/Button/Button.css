.button {
  border-radius: 12px;
  font-size: 18px;
  line-height: 27px;
  font-weight: 700;
  border: none;
  cursor: pointer;
  transition: all var(--animation);
  padding: 18px 61px 17px 61px;
  border: 2px solid var(--color-result);
  color: var(--color-result);
  background: none;
}

.button--primary {
  padding: 15px 36px 14px 37px;
  border: none;
  background-color: var(--color-hover);
  color: var(--color-white);
  box-shadow: 0px 2px 4px var(--color-shadow);
}

.button:hover {
  opacity: 0.7;
}
