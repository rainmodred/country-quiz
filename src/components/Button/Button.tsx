import React from 'react';
import './Button.css';

interface ButtonProps {
  children: React.ReactNode;
  primary?: boolean;
}

export default function Button({ children, primary }: ButtonProps) {
  const isPrimary = primary ? 'button--primary' : '';
  const classses = `button ${isPrimary}`;

  return (
    <button type="button" className={classses}>
      {children}
    </button>
  );
}
