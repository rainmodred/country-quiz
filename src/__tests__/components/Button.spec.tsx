import { render } from '@testing-library/react';
import Button from '../../components/Button/Button';

describe('Button component', () => {
  it('renders Button', () => {
    const { container, getByText } = render(<Button>button</Button>);

    expect(getByText(/button/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
