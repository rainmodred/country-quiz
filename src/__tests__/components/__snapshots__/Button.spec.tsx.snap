// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders Button 1`] = `
<div>
  <button
    class="button "
    type="button"
  >
    button
  </button>
</div>
`;
